### PR TITLE
chore(argo-cd): Add Proxy Extensions config

### PR DESCRIPTION
**Pull request by charlychiu**
Checklist:

* [X] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [X] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [X] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [X] Any new values are backwards compatible and/or have sensible default.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

The parameter `server.enable.proxy.extension` is already used in the original `charts/argo-cd/templates/argocd-server/deployment.yaml`.

### CODE REVIEW

**Review by yu-croco — DISMISSED** [review]
Thank you for your contribution. LGTM.

File: charts/argo-cd/values.yaml [hunk 1]
```diff
@@ -266,6 +285,8 @@ configs:
     server.disable.auth: false
     # -- Enable GZIP compression
     server.enable.gzip: true
+    # -- Enable proxy extension feature
```
Comment (mkilchhofer) [inline charts/argo-cd/values.yaml:288]: Can we mention the alpha phase here, please?

```suggestion
    # -- Enable proxy extension feature. (proxy extension is in Alpha phase)
```

**Review by mkilchhofer — COMMENTED** [review]
We can go with it (I have no strict opinion here). But normally this change is only a docs change since all released params from upstream (Deployment manifests with the optional Env mappings) are normally supported already.

Since it is a ALPHA feature, we should highlight it somewhere in the params docs as requested below.

File: charts/argo-cd/values.yaml (hunk 1 above)
Comment (charlychiu) [reply charts/argo-cd/values.yaml]: @mkilchhofer 

Thank you for your feedback. The changes have been committed. Please kindly review.

**Review by mkilchhofer — APPROVED** [review]

File: charts/argo-cd/values.yaml (hunk 1 above)
Comment (mkilchhofer) [reply charts/argo-cd/values.yaml]: @charlychiu thank you for implementing the propoal 👍

**Review by mbevc1 — APPROVED** [review]

**Pull request merged**